### PR TITLE
Add instructions for using Git in interactive sessions

### PR DESCRIPTION
**Pull request by aaasen**

### CODE REVIEW

File: docs/start/interactive.md [hunk 1]
```diff
@@ -73,3 +73,66 @@ Note: Some features, like user mapping, will not work with all images.
...
+are easy to generate and revoke. You can also authenticate using a private SSH key
+but be sure to generate one specifically for Beaker since these machines are shared
+across all of AI2.
+
+First, generate a new personal access token [here](https://github.com/settings/tokens/new)
+and grant it the repo scope.
+
+![Generating a personal access token with repo scope](/docs/images/github-personal-access-token-scope.png)
+
+You will only be able to view the on GitHub when it is created so store it in 1Password
```
Comment (ckarenz) [inline docs/start/interactive.md:89]:
```suggestion
You will only be able to view the token on GitHub when it is created so store it in 1Password
```

File: docs/start/interactive.md [hunk 2]
```diff
@@ -73,3 +73,66 @@ Note: Some features, like user mapping, will not work with all images.
 In the AllenNLP image, you will see a prompt like `I have no name!@fd82c7800efa:~$`.
 Please contact the Beaker team if you need help setting up a custom environment
 for your interactive sessions.
+
+## Git Authentication
+
+We recommend authenticating with GitHub using personal access tokens since they
+are easy to generate and revoke. You can also authenticate using a private SSH key
+but be sure to generate one specifically for Beaker since these machines are shared
+across all of AI2.
```
Comment (ckarenz) [inline docs/start/interactive.md:82]: This process seems pretty involved.  I'd lean toward the SSH key with a strong password.  It's a personal decision so I think we should present both options.

Comment (ckarenz) [reply docs/start/interactive.md:82]: Here's [GitHub's guide](https://docs.github.com/en/github/authenticating-to-github/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent)

File: docs/start/interactive.md [hunk 3]
```diff
@@ -73,3 +73,66 @@ Note: Some features, like user mapping, will not work with all images.
...
+
+![Generating a personal access token with repo scope](/docs/images/github-personal-access-token-scope.png)
+
+You will only be able to view the on GitHub when it is created so store it in 1Password
+for later use.
+
+Now, you should be able to clone any GitHub repository over HTTPS.
+When prompted for your password, enter the personal access token.
+
+### Credential Helper
```
Comment (ckarenz) [inline docs/start/interactive.md:95]: I recommend providing a Cliff's Notes version of these instructions (a single block of bash commands), a brief explanation of what it does, and link out to GitHub's docs for further reading.

Main rationale:
1. Some users are frustrated by long instructions (> 1 page) and we can save them time by boiling it all down.
2. It's tempting to associate added complexity with the tool and we have a hill to climb with regard to perception.
3. Documentation for external tooling can go stale very quickly. The less we write, the less we have to maintain.

File: docs/start/interactive.md (hunk 2 above)
Comment (aaasen) [reply docs/start/interactive.md:82]: I think this is less involved than authenticating with an SSH key. It's really just 3 steps:
1. Generate a new token.
2. Run `git config --global credential.helper store`
3. Type in the token next time you do a Git operation.

Maybe I can pare this down so it doesn't seem as complicated.

File: docs/start/interactive.md [hunk 4]
```diff
@@ -73,3 +73,66 @@ Note: Some features, like user mapping, will not work with all images.
 In the AllenNLP image, you will see a prompt like `I have no name!@fd82c7800efa:~$`.
 Please contact the Beaker team if you need help setting up a custom environment
 for your interactive sessions.
+
+## Git Authentication
```
Comment (ckarenz) [inline docs/start/interactive.md:77]: These instructions are really thorough.  Thank you for diving into this.

**Pull request merged**